### PR TITLE
[SW-1281] Fix bad representation of predictionCol on H2OMOJOModel

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOModel.scala [hunk 1]
```diff
@@ -77,7 +77,7 @@ class H2OMOJOModel(override val uid: String)
       case _ => throw new RuntimeException("Unknown model category")
     }
 
-    Seq(StructField(getOutputCol(), StructType(fields), nullable = false))
+    Seq(StructField(getPredictionCol, StructType(fields), nullable = false))
```
Comment (mn-mikke) [inline ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOModel.scala]: nit: ```getPredictionCol()```

File: ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOModel.scala [hunk 2]
```diff
@@ -163,7 +163,7 @@ class H2OMOJOModel(override val uid: String)
     val flattenedDF = H2OSchemaUtils.flattenDataFrame(dataset.toDF())
     val relevantColumnNames = flattenedDF.columns.intersect(getFeaturesCols())
     val args = relevantColumnNames.map(flattenedDF(_))
-    flattenedDF.select(col("*"), getModelUdf()(struct(args: _*)).as(getOutputCol()))
+    flattenedDF.select(col("*"), getModelUdf()(struct(args: _*)).as(getPredictionCol))
```
Comment (mn-mikke) [inline ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOModel.scala]: nit: ditto

File: ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOReader.scala [hunk 3]
```diff
@@ -18,13 +18,25 @@
...
   override def load(path: String): T  = {
-    val model = super.load(path)
+    val metadata = DefaultParamsReader.loadMetadata(path, sc)
+    val cls = Utils.classForName(metadata.className)
+    val instance =
+      cls.getConstructor(classOf[String]).newInstance(metadata.uid).asInstanceOf[Params]
+
+    val parsedParams = metadata.params.asInstanceOf[JsonAST.JObject].obj.map(_._1)
+    val allowedParams = instance.params.map(_.name)
+    val filteredParams = parsedParams.filter(!allowedParams.contains(_))
```
Comment (mn-mikke) [inline ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOReader.scala]: WDYT about using the```intersect``` method?

File: ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOReader.scala [hunk 4]
```diff
@@ -18,13 +18,25 @@
...
+    metadata.getAndSetParams(instance, Option(filteredParams))
```
Comment (mn-mikke) [inline ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOReader.scala]: nit: Can ```filteredParams``` be ```null```? If not, I would prefer to use ```Some(filteredParams)```.

File: ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOModel.scala (hunk 1 above)
Comment (jakubhava) [reply ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOModel.scala]: 👍

File: ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOReader.scala (hunk 4 above)
Comment (jakubhava) [reply ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOReader.scala]: yup - I think it can't be null

Comment (jakubhava) [reply ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOReader.scala]: setting to Some

File: ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOReader.scala (hunk 3 above)
Comment (jakubhava) [reply ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOReader.scala]: Thanks -> as per our discussion in the office, intersect is not a good fit, but `diff` is

**Pull request merged**